src/fu-tool.c

```diff
@@ -2576,6 +2606,12 @@ main (int argc, char *argv[])
 		     /* TRANSLATORS: command description */
 		     _("Convert a firmware file"),
 		     fu_util_firmware_convert);
+	fu_util_cmd_array_add (cmd_array,
+		     "firmware-build",
+		     "BUILDER-XML FILENAME-DST",
```

Comment:
I guess actually having the XML as an input means you should document the expected schema somewhere.

Reply:
It's tricky, as the schema will depend on the subclass of FuFirmware. For instance, FuBcm57xxFirmware has attributes `model` and `vendor`.

Reply:
Maybe a generic schema that would apply to everyone and leave optional fields as just that?

Reply:
Yup, that works. Any preference for XSD over RELAX NG, or do you just think an HTML document explaining the format is suitable? I could also do a _mega_-gtk-doc comment explaining it all in the existing generated documentation.

Reply:
I went with a mega-gtk-doc comment in the end, but I don't mind adding a schema too.

Reply:
I think the gtk documentation is fine, it will end up in fwupd.github.io (although I acknowledge you'll need to know how to find it).